components/TreeHeatmap/index.js

```diff
@@ -119,16 +124,20 @@ export default class TreeHeatmap extends VisComponent {
 
     let keys = Object.keys(this.data[0]);
     let idColumn = this.idColumn;
-    if (!idColumn) {
+    if (idColumn === undefined) {
       idColumn = keys.includes('_id') ? '_id' : idColumn;
       idColumn = keys.includes('_') ? '_' : idColumn;
       idColumn = keys.includes('') ? '' : idColumn;
     }
+    if (idColumn === undefined) {
+      console.log('TreeHeatmap: No suitable idColumn found.');
+      return;
+    }
 
     let rows = [];
     let reachedMetadata = false;
     this.data.forEach(row => {
-      let id = row[idColumn];
+      let id = '' + row[idColumn];
```

Comment:
I don't really know (see http://stackoverflow.com/questions/14918778/why-concat-empty-character-is-faster-than-tostring).

I view using `'' + x` for strings and `+x` for numbers as similar syntax to using `=>` for functions rather than `function` and binding-- it is not obvious to the novice, but is what the language has evolved to use and is common.